### PR TITLE
Rework depression/jolly quirk into brain trauma

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
balance: The Jolly and Depression quirks are far more active in their
mood buff/debuffs. You should expect the mood change around half the
time.
balance: Since the Depression quirk is now more severe, it is now worth
2 hardcore random points.
tweak: The Jolly and Depression quirks are now unhealable brain trauma,
similar to how the paraplegic quirk also leaves an unhealable brain
trauma. These traumas are not in the regular trauma pool, and cannot be
gained outside quirks.
/:cl:

Instead of the depression and jolly effect being limited to quirks, they
are now implemented as brain traumas, allowing their use elsewhere, and
moves quirks to be more consistently integrated into the rest of the
system.

Their current implementation mean that a round could pass without a
single activation of depression/jolly mood buff, so it needed to be
changed.

- TRAIT_JOLLY and TRAIT_DEPRESSION as traits have been removed, as they
  had no real meaning in terms of anything checking for those traits.
- Gonbolas now have a different positive mood message from before, but
  function in the same way.

### CODE REVIEW

**Comment by optimumtact** [comment]
>Instead of the depression and jolly effect being limited to quirks, they
are now implemented as brain traumas, allowing their use elsewhere, and
moves quirks to be more consistently integrated into the rest of the
system.

Stupid

File: code/datums/brain_damage/quirk.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Quirk brain traumas are ungainable through play, but a player may start
+// with a number of them through quirks.
+
+/datum/brain_trauma/quirk
```
Comment (Jared-Fogle) [inline code/datums/brain_damage/quirk.dm:4]: Why do we need an entire subtype for this?

**Comment by ATH1909** [comment]
don't really see why you had to convert these into brain traumas

if anything, quirks should be LESS dependent on brain traumas, not more dependent on them (what if we add a brainless species in the future???)

**Comment by Pugie** [comment]
I never knew being jolly made you clinically retarded, thanks for the heads up!

**Comment by coiax** [comment]
Working on this, I realise what I really want to do is just "tie quirks together with other systems", rather than just wed all of them to brain traumas when it doesn't fit.

I have a good idea of how to rework this.

**Pull request closed without merging**